### ISSUE TITLE
Merge operations leads to incorrect amount on bank accounts

### ISSUE THREAD

**Issue by Flore-mozfr** [bnjbvr/kresus#245]
When merging operations, the amount of one of the operations is deducted from the total amount on the bank account.
example : 
- My credit card account for the month is -500€ (correct amount shown on my bank website)
- 1 card transaction (100€) was imported twice by Kresus
- I merge these 2 transactions to clean up everything
- Kresus now shows my credit card account as -400€

Problem: it's totally wrong and Kresus will never correct that. There will always be a 100€ discrepancy between the real amount and the amount calculated by Kresus. It will be even worse if other operations are merged.

One way to work around this would be if Kresus would sync the amounts on the bank accounts and not only the operations. This way any discrepancy would be corrected.

My bank is société générale.

**Comment by ZeHiro** [comment]
This is exacly the expected behaviour. 
This function is here to cover the case where kresus imports twice the same operation based on your bank website. Usually, some pending operations has a generic label ("Operation Carte" for my bank, LCL); and is then updated with the real label. If kresus imports these two operations, the balance will be incorrect (-600 instead of -500). Merging the operations will allow to provide the right balance.
The duplicate detection function is not perfect and needs to be used carefully (many false duplicates). In a next version this should be improved (#183). And merging operations will some day be revertible.  About recyncing the balance, it is on the roadmap also #190.

**Comment by bnjbvr** [comment]
Thanks @Flore-mozfr  for opening the issue, and many thanks to @ZeHiro for explaining the current state of things. Flore, is this an operation that Kresus detects as a duplicate but which is actually NOT a duplicate (i.e. false positive)?

**Comment by Flore-mozfr** [comment]
No, it is a real duplicate. I synced my account and a few were imported twice (same amount, same precise description, same ID). I saw it immediately on the list of operations, that's why I ran the doublon check and merged the operations.
And for each doublon merged, the balance on my account was updated by adding the amount of one of the duplicate.

The thing is the initial balance was correct (provided by my bank), the duplicate operations were just a display glitch. But when I merged the amount was wrong. And as there is no sync, the amount will be wrong ad vitam aeternam, because Kresus will just add future operations.

I have a monthly credit card (carte bleue à débit différé), so it is a separate account where all credit card operations are applied, it has always a negative balance. And at the end of the month (when everything is transferred to the global account), there is an operation that adds the exact amount of all expenses allowing the balance to return to zero to start the new month.

**Comment by bnjbvr** [comment]
So the issue is that some operations were imported several times on the first import? Wow, this sounds crazy. I'll look carefully at the code to see how this case can happen, but the initial data ought to be trusted...

One thing I don't clearly understand: how is that related to your credit card account? Is it that all the credit card account operations happen on your main account? Or something else?

**Comment by Flore-mozfr** [comment]
I have a separate account for my credit card where all card operations are accounted for throughout the month. On 1st day of each month, all these operations are transferred to the main account and the credit card account is credited with the same amount to put the balance back to zero.

For example, it would be : 
10/09 : card : -100 
12/09 : card : -25
14/09 : card : -25
15/09 : card : -50
24/09 : card : -20
30/09 : card : -30

On 30/09 the balance on this account is -250
And on 01/10, all these operations are transferred to my main account and +250 is added to the credit card. So balance is 0.

Now the problem was, I imported my account on 25/09 (for example). Balance was 220. But Kresus imported twice the 10/09 : card : -100 operation. I merged these 2 operations and Kresus added +100 to my balance. On Kresus, the balance on 25/09 is -120. It will continue to add and substract everything but this +100 discrepancy will remain.

**Comment by ZeHiro** [comment]
This one occured again. If deleting the duplicate operation manually, the balance will be messed up. (because of how initial balance is computed).
